haskell-src/Concordium/Types/Execution.hs

```diff
@@ -1967,6 +1967,13 @@ data RejectReason = ModuleNotWF -- ^Error raised when validating the Wasm module
                   | PoolWouldBecomeOverDelegated
                   -- |The pool is not open to delegators.
                   | PoolClosed
+                  -- |The module reference provided for the contract upgrade did not
+                  -- point to a deployed module.
+                  | UpgradeInvalidModuleReference !ModuleRef
```

Comment:
I don't think these will be used, at least not until we have stack traces for failed executions.
The reason being that when a contract call `upgrade` it will be told whether it failed and should decide how to deal with the failure. In the case it rejects the transaction, it will be a `RejectedInit` or `RejectedReceive`.

Reply:
That is correct. We will need this in the internals, i.e., signalling from the scheduler back to the contract, but we won't record it in the final transaction event.